client/src/context/useAuthContext.tsx

```diff
@@ -22,18 +22,9 @@ export const AuthProvider: FunctionComponent = ({ children }): JSX.Element => {
   const [loggedInUser, setLoggedInUser] = useState<User | null | undefined>();
   const history = useHistory();
 
-  const updateLoginContext = useCallback(
-    (data: AuthApiDataSuccess) => {
-      if (!data.user.username) {
-        setLoggedInUser(data.user);
-        history.push('profile_settings');
-      } else {
-        setLoggedInUser(data.user);
-        history.push('/dashboard');
-      }
-    },
-    [history],
-  );
+  const updateLoginContext = useCallback((data: AuthApiDataSuccess) => {
```

Comment:
Has there been a performance concern that prompts us to utilize `useCallback` here. By doing this there is more work being done behind the scenes.

Reply:
Hey Ethan, this was a part of the starter code but I suspect it is because we're utilizing useEffect. If we were to remove it, I believe we would get infinite re-renders if we let useEffect handle it w/o a callback.